### PR TITLE
Implementation of Save_history, Get_history, and Clear use cases

### PR DESCRIPTION
**Pull request by JG808**

### CODE REVIEW

File: src/app/Main.java [hunk 1]
```diff
@@ -33,7 +34,10 @@ public static void main(String[] args) {
         FileSearchDataAccessObject searchDataAccessObject;
         searchDataAccessObject = new FileSearchDataAccessObject(new CommonPlaceFactory(), "src/data_access/filters.csv", "listingJSON.json");
 
-        SearchView searchView = SearchUseCaseFactory.create(viewManagerModel, searchViewModel, listingResultsViewModel, searchDataAccessObject);
+        HistoryDataAccessObject historyDataAccessObject;
+        historyDataAccessObject = new HistoryDataAccessObject("history.csv");
+
+        SearchView searchView = SearchUseCaseFactory.create(viewManagerModel, searchViewModel, listingResultsViewModel, searchDataAccessObject, historyDataAccessObject, historyDataAccessObject, historyDataAccessObject);
```
Comment (Arnnav-A) [inline src/app/Main.java:40]: Is there a way that can avoid passing the same argument 3 times?

**Review by Arnnav-A — CHANGES_REQUESTED** [review]
Get history controller and save controller contain unnecessary throws exception in method signature

File: src/entity/Listing.java [hunk 2]
```diff
@@ -9,4 +9,6 @@ public interface Listing {
     String getCity();
 
     String getFilter();
+
+    Place getPlaceFromName(String name);
```
Comment (nicolasdiasm) [inline src/entity/Listing.java:13]: This method is not necessary anymore

File: src/entity/CommonListing.java [hunk 3]
```diff
@@ -29,6 +29,7 @@ public String getFilter() {
         return filter;
     }
 
+    @Override
```
Comment (nicolasdiasm) [inline src/entity/CommonListing.java:32]: This method is not useful anymore, as the implementation of the placeInfo use case was completely reformulated

File: src/view/SearchView.java [hunk 4]
```diff
@@ -25,8 +28,12 @@ public class SearchView extends JPanel implements ActionListener, PropertyChange
     private final SearchController searchController;
     // create button for search.
     final JButton search;
+    final JButton getHistory;
+    final JButton clearHistory;
```
Comment (nicolasdiasm) [inline src/view/SearchView.java:31]: add controllers as instance attributes, just like we did with searchController

File: src/view/SearchView.java [hunk 5]
```diff
@@ -25,8 +28,12 @@ public class SearchView extends JPanel implements ActionListener, PropertyChange
...
 
-    public SearchView(SearchViewModel searchViewModel, SearchController searchController) {
+
+    public SearchView(SearchViewModel searchViewModel, SearchController searchController, SaveController saveController,
+                      GetHistoryController getHistoryController, ClearController clearController) {
```
Comment (nicolasdiasm) [inline src/view/SearchView.java:35]: add
this.(name of the controller) = (name of the controller);
for each controller now added as instance attributes

File: src/interface_adapter/clear/ClearPresenter.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    public ClearPresenter(SearchViewModel searchViewModel, ViewManagerModel viewManagerModel) {
+        this.searchViewModel = searchViewModel;
+        this.viewManagerModel = viewManagerModel;
+    }
+
+    @Override
+    public void prepareView(ClearOutputData response) {
+        searchViewModel.firePropertyChanged();
+        viewManagerModel.firePropertyChanged();
+        response.message();
```
Comment (nicolasdiasm) [inline src/interface_adapter/clear/ClearPresenter.java:21]: Related to my comment in the output data of the clear use case

**Review by nicolasdiasm — CHANGES_REQUESTED** [review]
add controllers as instance attributes, just like we did with searchController

**Review by Arnnav-A — APPROVED** [review]

**Review by nicolasdiasm — APPROVED** [review]
We can make this changes after merging it as they are mostly related to clean architecture

**Pull request merged**